central/role/datastore/datastore_impl.go

```diff
@@ -269,6 +277,14 @@ func (ds *dataStoreImpl) RemovePermissionSet(ctx context.Context, id string) err
 	return nil
 }
 
+func verifyPermissionSetOriginMatches(ctx context.Context, ps *storage.PermissionSet) error {
+	if !declarativeconfig.IsOriginModifiable(ctx, ps.GetTraits().GetOrigin()) {
+		return errors.Wrapf(errox.InvalidArgs, "permission set %q's origin is %s, cannot be modified or deleted within this context",
```

Comment:
Do you think we should leak this "within this context" to any potential clients?
If someone calls this via API, I can imagine that it may be confusing.

Maybe we should just do the following:
- Return `errox.NotAuthorized` to clearly specify that its not invalid args, and in the current context of trying to delete via API there's no "valid" args to give, it's simply not authorized.
- As an error message, swap out the "within this context" with "with the current permission"